Change color for added menu items and update hover icon (in Customizer)

zerosonesfun opened the pull request:
This change makes the added menu item in Customizer WCAG compliant while leaving subtle effects to let the user know that they already added the menu item, but also, if they want they can add it again. This decision was made because it's the least amount of CSS changes while still making sense.

The same dark gray color used initially for menu item labels in Customizer is reused because a color change is not needed. The user will already see the checkmark icon indicating that the item was successfully added. Since you can add the same menu item multiple times (which should stay as-is in case a user has a complex repetitive menu or some reason I'm not thinking of), I also changed the on hover icon to the plus icon. This way, after adding, the user sees the check mark (☑️ great, it added!). If they want to re-add, instead of seeing the initial blue hover color, they see the dark gray _but_ also the plus icon. This combination says, "you already added this, but add it again if you want."

Other solutions would add more lines of CSS or require more changes. I'm a fan of fixing the WCAG issue with minimal changes.

Trac ticket: https://core.trac.wordpress.org/ticket/64013

---

Current:
<img width="1057" height="349" alt="Screenshot 2025-11-23 at 10 21 01 AM" src="https://github.com/user-attachments/assets/31345504-f45e-409e-9e81-0d4a30048536" />

New, after adding:
<img width="1057" height="349" alt="Screenshot 2025-11-23 at 10 37 02 AM" src="https://github.com/user-attachments/assets/02dc3303-6a8a-459b-8278-54594afcb95a" />

New, after adding and on hover:
<img width="1057" height="349" alt="Screenshot 2025-11-23 at 10 37 16 AM" src="https://github.com/user-attachments/assets/3a9da374-7cd6-4290-ae29-7430dab635c1" />

zerosonesfun committed:
Change color for added menu items and update hover icon

https://core.trac.wordpress.org/ticket/64013

This change makes the added menu item in Customizer WCAG compliant while leaving subtle effects to let the user know that they already added the menu item, but if they want they can add it again.

sabernhardt commented on src/wp-admin/css/customize-nav-menus.css:
```diff
@@ -665,13 +665,17 @@
 #available-menu-items .menu-item-handle.item-added:hover .item-add,
 #available-menu-items .menu-item-handle.item-added .item-add:focus {
-	color: #8c8f94;
+	color: #1d2327;
```
```suggestion
	color: #646970;
```

`#646970`, or [Gray 50](https://codepen.io/ryelle/full/WNGVEjw), would have sufficient contrast of 5.53:1 against white. It also matches the shade used for the item type (Page, Custom Link, etc.).

<img width="599" height="298" alt="Added menu item with Gray 50" src="https://github.com/user-attachments/assets/df181df8-a759-410d-8ab3-3d63ab7eb0dc" />

I prefer to follow the original intent of making the added items a little lighter. If changing all the text to gray does not improve the interface, however, the entire ruleset could be removed instead of matching the darker gray. Then the menu item would still have the lighter text for the item type and blue text when hovering over the menu item.

sabernhardt commented on src/wp-admin/css/customize-nav-menus.css:
```diff
@@ -665,13 +665,17 @@
 	content: "\f147" / '';
 }
 
+#available-menu-items .menu-item-handle.item-added:hover .item-add:before {
```
```suggestion
#available-menu-items .menu-item-handle.item-added:hover .item-add:before,
#available-menu-items .menu-item-handle.item-added .item-add:focus:before {
```

I am not convinced that the icon should change on `:hover`, but replacing the button icon on `:focus` could help more. This suggestion applies the plus icon to both `:hover` and `:focus` states.

Plus icon on `:focus` (with Gray 50 for text):
<img width="304" height="174" alt="Plus icon on focus" src="https://github.com/user-attachments/assets/9853c80b-7a13-435e-97a2-49f30dea9caf" />

Hover:
<img width="304" height="174" alt="Plus icon on hover" src="https://github.com/user-attachments/assets/a5e81f76-64c6-470c-886d-7529f7d3ebc9" />

joedolson committed:
Apply suggestions from code review

Co-authored-by: Stephen A. Bernhardt <sabernhardt@yahoo.com>

joedolson committed:
Remove hover and focus styles for item-added

Removed additional hover and focus icon changes for added menu items.

Avoiding scope creep on the issue.

showravhasan1 commented:
## Test Report ✅
Tested on a local WordPress 6.9 installation with Twenty Twenty-One theme.

### Bug Confirmed
- **Current color:** `#8c8f94` 
- **Contrast ratio:** 3.24:1 (fails WCAG AA 4.5:1 minimum)
- Added menu items in Customizer are hard to read

### Steps to Reproduce
1. Appearance → Customize → Menus
2. Open "Primary Menu"
3. Click "Add Items" → Pages
4. Add any page to the menu
5. Observe: The added item text is low contrast gray

### Verdict
LGTM 👍 The fix to darken the text color correctly addresses the accessibility issue.

Also left a comment on the Trac ticket: https://core.trac.wordpress.org/ticket/64013
